src/fides/api/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -1198,8 +1253,140 @@ def _deny_request(
+        return bool(
+            PreApprovalWebhookReply.filter(
+                db=db,
+                conditions=(
+                    (PreApprovalWebhookReply.webhook_id == webhook_id)
+                    & (PreApprovalWebhookReply.privacy_request_id == privacy_request_id)
+                ),
+            ).first()
+        )
+
+    # Check if not all replies have been received
+    if not all(_reply_exists_for_webhook(webhook.id) for webhook in all_webhooks):
+        # Subtlety here is that if a new webhook is configured between when the privacy request was created and when
+        # the last webhook replied, then the privacy request can never be automatically approved. It must instead be
+        # manually approved in the Admin UI.
+        logger.info(
+            "Not all pre-approval webhooks have responded for privacy request '{}'. Cannot automatically approve request.",
+            privacy_request_id,
+        )
+        return
+    # Check if all replies are true
+    replies_for_privacy_request = PreApprovalWebhookReply.filter(
+        db=db,
+        conditions=(PreApprovalWebhookReply.privacy_request_id == privacy_request_id),
+    ).all()
+    if not all(reply.is_eligible for reply in replies_for_privacy_request):
+        logger.info(
+            "Not all pre-approval webhooks have responded with eligible for privacy request '{}'. Cannot automatically approve request.",
+            privacy_request_id,
+        )
+        return
+
+    logger.info(
+        "All pre-approval webhooks have responded with eligible for privacy request '{}'. Proceeding to automatically approve request.",
+        privacy_request_id,
+    )
+    review_privacy_request(
+        db=db,
+        request_ids=[privacy_request_id],
+        config_proxy=config_proxy,
+        process_request_function=_approve_request,
+        webhook_id=webhook.id,
+        user_id=None,
+    )
+
+
+@router.post(
+    PRIVACY_REQUEST_PRE_APPROVE_NOT_ELIGIBLE,
+    status_code=HTTP_200_OK,
+)
+def mark_privacy_request_pre_approve_not_eligible(
+    privacy_request_id: str,
+    *,
+    db: Session = Depends(deps.get_db),
+    webhook: PreApprovalWebhook = Security(
+        verify_callback_oauth_pre_approval_webhook, scopes=[PRIVACY_REQUEST_REVIEW]
+    ),
+) -> None:
+    """Marks privacy request as not eligible for automatic approval, regardless of what other webhook responses we receive"""
+    _validate_privacy_request_pending_or_error(db, privacy_request_id)
+    logger.info(
+        "Marking privacy request '{}' as not eligible for automatic approval via webhook '{}' for connection config '{}'.",
+        privacy_request_id,
+        webhook.key,
+        webhook.connection_config.key,
+    )
+    PreApprovalWebhookReply.create(
+        db=db,
+        data={
+            "webhook_id": webhook.id,
+            "privacy_request_id": privacy_request_id,
+            "is_eligible": False,
+        },
     )
```

Comment:
I'd also add some sort of handling here so we return a proper HTTP exception to the end user